### PR TITLE
Make fastd syslog level configurable

### LINKED ISSUE #896: wanted : fastd syslog level in site.conf (or stop flooding syslog)
i assumed that that i could set syslog level in fastd - 
i'd love to set it in site conf - without a hacky way 

while we have plenty Gateway Servers, our syslog become pretty unreadable with all the fastdinfo - it is limited in (ring)size - and by hunting down other problems the 24/7 running fastd even in mesh only routers spam the syslog a lot 
for us 
uci set fastd.mesh_vpn.syslog_level='warn'
or in site conf under fastd mesh-vpn ... a syslog_level = 'warn' 
is enough, if we need more info in detail we could do it seperatly and easily on specific routers

### PR DESCRIPTION
**Pull request by viisauksena**
it is easy to set this to a higher value again if needed. This will prevent fastd from flooding the log ringbuffer, even when it is not needed/no uplink via fastd. 
This log - flooding raise with every peer known from site.conf.

Fixes #896

### CODE REVIEW

**Comment by viisauksena** [comment]
if i check if syslog_level is set in mesh_vpn similar to the enabled stuff in line 18
```local syslog_level = uci:get('fastd', 'mesh_vpn', 'syslog_level')
if not syslog_level then
  syslog_level = 'warn'
end
```
then it is sufficient to set line 55 from 
	syslog_level = 'warn',
to 
	syslog_level = site.fastd_mesh_vpn.syslog_level,
am i right with this? and if so, i need to add some lines to the doku right?

**Comment by NeoRaider** [comment]
Hmm, I'd rather increase the ring buffer size, as fastd connection issues are a common problem, so having the connection attempts in the log is helpful.

**Comment by viisauksena** [comment]
you can guess yourself, having 10 gw , with limit 2 will fill this buffer very fast (even if bigger) - since the connection attempt appear even if limit is fullfilled or no fastd/uplink is available at all
an event which occurs some hours ago is then lost

**Comment by viisauksena** [comment]
if my tests were right - this should do the trick to have it configurable while keeping your default (verbose) if not set explicitly in site.conf

File: package/gluon-mesh-vpn-fastd/luasrc/lib/gluon/upgrade/400-mesh-vpn-fastd [hunk 1]
```diff
@@ -21,6 +21,12 @@ if not enabled then
 end
 
 
+local syslog_level = uci:get('fastd', 'mesh_vpn', 'syslog_level')
+if not syslog_level then
+  syslog_level = 'verbose'
+end
```
Comment (NeoRaider) [inline package/gluon-mesh-vpn-fastd/luasrc/lib/gluon/upgrade/400-mesh-vpn-fastd]: This could be simplified to
```lua
local syslog_level = uci:get('fastd', 'mesh_vpn', 'syslog_level') or 'verbose'
```

File: docs/user/site.rst [hunk 2]
```diff
@@ -182,12 +182,15 @@ fastd_mesh_vpn
     In any case, the ``null`` method should always be the first method in the list
     if it is supported at all. You should only set `configurable` to `true` if the
     configured peers support both the ``null`` method and methods with encryption.
+    
+    You can set syslog_level from verbose(default) to warn to reduce syslog output.
```
Comment (NeoRaider) [inline docs/user/site.rst]: Please add a space before the parentheses.

**Comment by NeoRaider** [comment]
This looks mostly okay, please also add the new optoin to the site check script to ensure only valid values can be set.

**Comment by viisauksena** [comment]
i am not sure in all allowed values while fastd seem not to allow the full scale of possible debug values - i remember having some trouble with them by manualy setting them. 
I guess you mean https://github.com/freifunk-gluon/gluon/blob/master/scripts/check_site_lib.lua -  i didnt get where/how to ensure valid values there

**Comment by NeoRaider** [comment]
You'd add a line like the following to https://github.com/freifunk-gluon/gluon/blob/master/package/gluon-mesh-vpn-fastd/check_site.lua
```lua
need_array_of('fastd_mesh_vpn.syslog_level', {'error', 'warn', 'info', 'verbose', 'debug', 'debug2'}, false)
```

**Comment by NeoRaider** [comment]
Eeeeh, not `need_array_of`, I mean `need_one_of`

**Comment by viisauksena** [comment]
ich mach das heut nacht mal fertig

**Comment by viisauksena** [comment]
done ! - 
thx for help

File: package/gluon-mesh-vpn-fastd/luasrc/lib/gluon/upgrade/400-mesh-vpn-fastd (hunk 1 above)
Comment (viisauksena) [inline package/gluon-mesh-vpn-fastd/luasrc/lib/gluon/upgrade/400-mesh-vpn-fastd]: if you prefer your simplification , i could change it to your suggestion

Comment (NeoRaider) [inline package/gluon-mesh-vpn-fastd/luasrc/lib/gluon/upgrade/400-mesh-vpn-fastd]: That would be great, it would also make the coding style match other scripts.

File: package/gluon-mesh-vpn-fastd/luasrc/lib/gluon/upgrade/400-mesh-vpn-fastd [hunk 3]
```diff
@@ -15,11 +15,9 @@ users.remove_user('gluon-fastd')
 users.add_group('gluon-fastd', 800)
 
 
-local enabled = uci:get('fastd', 'mesh_vpn', 'enabled')
-if not enabled then
-  enabled = site.fastd_mesh_vpn.enabled and 1 or 0
-end
+local enabled = uci:get('fastd', 'mesh_vpn', 'enabled') and 1 or 0
```
Comment (NeoRaider) [inline package/gluon-mesh-vpn-fastd/luasrc/lib/gluon/upgrade/400-mesh-vpn-fastd]: Please don't change unrelated code - in this case, you've even broken it by dropping `site.fastd_mesh_vpn.enabled`, which is what the whole `and 1 or 0` is about.

Comment (viisauksena) [inline package/gluon-mesh-vpn-fastd/luasrc/lib/gluon/upgrade/400-mesh-vpn-fastd]: mhm, tested on a random node - seemed to work fine - but i can delete this part

Comment (viisauksena) [inline package/gluon-mesh-vpn-fastd/luasrc/lib/gluon/upgrade/400-mesh-vpn-fastd]: mhm, i didnt get the problem - but ok so far - i reverted it.
